examples_utils/benchmarks/command_utils.py

```diff
@@ -189,49 +191,99 @@ def formulate_benchmark_command(
     return cmd
 
 
-def get_poprun_hosts(cmd: list) -> list:
+def get_num_ipus(benchmark_name: str) -> int:
+    num_ipus = re.findall(pattern="pod(\d+)", string=benchmark_name)
+    if len(num_ipus) < 0:
+        err = (f"Processing benchmark variant: {benchmark_name}."
+               " Malformed name. Benchmark must specify the number of IPUs "
+               "to be used in the form "
+               "[alphanumeric chars and underscore]+pod(\d+)"
+               "[alphanumeric_chars and underscore]+")
+        logger.error(err)
+        raise ValueError(err)
+    else:
+        num_ipus = num_ipus[0]
+        return num_ipus
+
+
+def query_option_in_cmd(cmd: list, option: list):
+    try:
+        index = cmd.index([x for x in cmd if any(opt in x for opt in option)][0])
+        return index
+    except:
+        return None
+
+
+def get_poprun_config(args, cmd) -> Dict:
+    """Get a poprun configuration dict storing key,value pairs of poprun options 
+    and supplied arguments. We store only a subset of options that are going 
+    to be needed in order to ensure that the job submission environment is correct:
+
+    Args: 
+       args (argparse.Namespace): Arguments passed to run the benchmarks
+            with        
+        cmd: (list): benchmark variant command
+    Return:
+        poprun configuration (Dict): key value pairs of poprun options and arguments    
+    """
+    poprun_config = {}
+
+    # Find where in the command list "poprun" and "python" exist
+    # If poprun is not called, then it cannot be multihost + multi-instance
+    poprun_index = query_option_in_cmd(cmd, ["poprun"])
+    if poprun_index is None:
+        return poprun_config
+
+    # Watch out for "python" instead of "python3"
+    python_index = query_option_in_cmd(cmd, ["python3", "python"])
+
+    parser = ArgumentParser()
+
+    configuration_options = [["-H", "--host"], ["--num-instances"], ["--vipu-server-host"], ["--vipu-server-port"],
```

Comment:
Is this confirmed to be the full set that we need? Or is this everything that poprun offers? If the latter, might be better to get this from `poprun --help` or something? 

I mean it works, just hoping that we dont need to update this everytime poprun CLI changes slightly (also unlikely)

Reply:
These are the only options that we need to interact with before submitting the job

Reply:
Ok well if its a restriced set then it makes sense.